File: backend/src/main/java/com/talentbridge/service/AdminService.java
package com.talentbridge.service;

import com.talentbridge.dto.AdminStatsDto;
import com.talentbridge.dto.JobResponse;
import com.talentbridge.dto.UserSummaryDto;
import com.talentbridge.entity.JobStatus;
import com.talentbridge.entity.Role;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface AdminService {

    AdminStatsDto getPlatformStats();

    Page<UserSummaryDto> getAllUsers(Role role, Boolean isActive, String search, Pageable pageable);

    UserSummaryDto updateUserStatus(Long userId, boolean active);

    Page<JobResponse> getAllJobs(JobStatus status, Pageable pageable);

    JobResponse updateJobStatus(Long jobId, JobStatus status);

    void deleteJob(Long jobId);
}

File: frontend/src/services/adminService.js
import api from './api';

export const adminService = {
  getStats: async () => {
    const response = await api.get('/admin/stats');
    return response.data;
  },

  getUsers: async (params = {}) => {
    const response = await api.get('/admin/users', { params });
    return response.data;
  },

  updateUserStatus: async (userId, active) => {
    const response = await api.put(`/admin/users/${userId}/status`, { active });
    return response.data;
  },

  getJobs: async (params = {}) => {
    const response = await api.get('/admin/jobs', { params });
    return response.data;
  },

  updateJobStatus: async (jobId, status) => {
    const response = await api.put(`/admin/jobs/${jobId}/status`, { status });
    return response.data;
  },

  deleteJob: async (jobId) => {
    const response = await api.delete(`/admin/jobs/${jobId}`);
    return response.data;
  },
};

export default adminService;
